conanfile.py

```diff
@@ -243,73 +232,106 @@ def imports(self):
-
-        cmake = CMake(self, set_cmake_flags=True, generator=self.generator)
-        if self.settings.compiler == "Visual Studio":
-            cmake.definitions["CONAN_LINK_RUNTIME_MULTI"] = cmake.definitions["CONAN_LINK_RUNTIME"]
-            cmake.definitions["CONAN_LINK_RUNTIME"] = False
-        cmake.definitions["BUILD_OPNAV"] = self.options.opNav
-        cmake.definitions["BUILD_VIZINTERFACE"] = self.options.vizInterface
-        cmake.definitions["EXTERNAL_MODULES_PATH"] = self.options.pathToExternalModules
-        cmake.definitions["PYTHON_VERSION"] = f"{sys.version_info.major}.{sys.version_info.minor}"
+    def generate(self):
+        if self.options.get_safe("pathToExternalModules"):
+            print(statusColor + "Including External Folder: " + endColor + str(self.options.pathToExternalModules))
 
-        if self.options.pyLimitedAPI != "":
-            cmake.definitions["PY_LIMITED_API"] = self.options.pyLimitedAPI
+        if self.settings.build_type == "Debug":
+            self.output.warning("Build type is set to Debug. Performance will be significantly slower.")
+
+        # -------------------------------------------------------------
+        # Run the CMake configuration generators.
+        # -------------------------------------------------------------
+        deps = CMakeDeps(self)
+        deps.set_property("eigen", "cmake_target_name", "Eigen3::Eigen3")   # XXX: Override, original is "Eigen3::Eigen"
+        deps.set_property("cppzmq", "cmake_target_name", "cppzmq::cppzmq")  # XXX: Override, original is "cppzmq"
+        deps.generate()
+
+        tc = CMakeToolchain(self)
+        generatorString = str(self.options.get_safe("generator"))
+        if generatorString == "":
+            # Select default generator supplied to cmake based on os
+            if self.settings.os == "Macos" and not self.options.get_safe("buildProject"):
+                generatorString = "Xcode"
+                tc.generator = generatorString
+            elif self.settings.os == "Windows":
+                generatorString = "Visual Studio 16 2019"
+                tc.generator = generatorString
+                self.options["*"].shared = True
+            else:
+                print("Creating a make file for project. ")
+                print("Specify your own using the -o generator='Name' flag during conan install")
+        else:
+            tc.generator = generatorString
+            if self.settings.os == "Windows":
+                self.options["*"].shared = True
+        print("cmake generator set to: " + statusColor + generatorString + endColor)
 
+        tc.cache_variables["BUILD_OPNAV"] = bool(self.options.get_safe("opNav"))
+        tc.cache_variables["BUILD_VIZINTERFACE"] = bool(self.options.get_safe("vizInterface"))
+        if self.options.get_safe("pathToExternalModules"):
+            tc.cache_variables["EXTERNAL_MODULES_PATH"] = self.externalModulesPath.resolve().as_posix()
```

Comment:
I think this should be `self.options.pathToExternalModules.resolve().as_posix()` instead

Reply:
Good catch.  This was one build scenario I didn't check.  I'll test on my computer and upload a fix to the branch.